docs: add doc on how to try a coder-preview build

EdwardAngert opened the pull request:
combined the steps I used into one doc to make it easier for everyone else who wants to deploy a quick coder-preview build

[preview](https://coder.com/docs/@install-coder-preview-quickly/install/coder-preview-docker)

EdwardAngert committed:
add coder-preview doc

EdwardAngert committed:
specify linux and mac

EdwardAngert committed:
clarify latest preview url

matifali commented:
Is the motive is to encourage more users to install the preview build?

bpmct commented:
This feels a bit strange to be first-class in our docs. I know other products have nightly/preview builds but tend to have them more hidden in the docs versus under the Install nav.

bpmct commented on docs/manifest.json:
```diff
@@ -96,6 +96,12 @@
 					"path": "./install/uninstall.md",
 					"icon_path": "./images/icons/trash.svg"
 				},
+				{
+					"title": "Install and test coder-preview",
+					"description": "Test the latest coder-preview build",
+					"path": "./install/coder-preview-docker.md",
+					"icon_path": "./images/icons/contributing.svg"
+				},
```
How does this look in the docs sidebar? It's a pretty long title

EdwardAngert replied:
it's not great - and it might not be totally accurate. I don't think we actually install Coder with the `docker run` command

I'll update it in a little bit

matifali replied:
IMO, it should live as a notice of type "Tip" on the docker install page. And not a separate page.

EdwardAngert commented:
@matifali 

> Is the motive is to encourage more users to install the preview build?

not really - it should be more of a "you could do this if you need to" and not something people think is an option. I think it actually belongs more in a troubleshooting section - would that make more sense?

matifali commented:
> not really - it should be more of a "you could do this if you need to" and not something people think is an option. I think it actually belongs more in a troubleshooting section - would that make more sense?

> [!TIP]
> IMO, it should live as a Tip on the docker install page. And not a separate page.

matifali reviewed (changes requested):
Blocking to prevent an accident merge.

matifali committed:
Merge branch 'main' into install-coder-preview-quickly

matifali committed:
Move preview installation to Docker section

- Consolidates preview release installation guidance under the Docker section.
- Removes the standalone page for coder-preview installation.

matifali committed:
Delete docs/install/coder-preview-docker.md
